src/zeit/retresco/search.py

```diff
@@ -0,0 +1,40 @@
+import elasticsearch
+import json
+import zeit.cms.interfaces
+import zeit.cms.result
+import zeit.retresco.interfaces
+import zope.interface
+
+
+@zope.interface.implementer(zeit.retresco.interfaces.IElasticsearch)
+class Elasticsearch(object):
+    """Search via Elasticsearch."""
+
+    def __init__(self, url):
+        self.client = elasticsearch.Elasticsearch([url])
+
+    def search(self, query, sort_order, start=0, rows=25):
+        """Search using `query` and sort by `sort_order`."""
+        query = query.copy()
+        query['_source'] = 'url'
+        __traceback_info__ = query
+        config = zope.app.appsetup.product.getProductConfiguration(
```

Comment:
Da es nur 2 Settings sind, würde ich die in der Factory-Funktion aus der Product-Config holen und dem Utility im Constructor übergeben. Auf jeden Fall sollte man es nicht auf zwei Stellen verteilen. ;)